pkg/ddl/tests/partition/multi_domain_test.go

```diff
@@ -246,19 +256,304 @@ func TestMultiSchemaDropListColumnsDefaultPartition(t *testing.T) {
+				"  `b` varchar(255) DEFAULT NULL,\n" +
+				"  PRIMARY KEY (`a`) /*T![clustered_index] CLUSTERED */,\n" +
+				"  UNIQUE KEY `idx_b_global` (`b`) /*T![global_index] GLOBAL */\n" +
+				") ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin\n" +
+				"PARTITION BY RANGE (`a`)\n" +
+				"(PARTITION `p0` VALUES LESS THAN (100),\n" +
+				" PARTITION `p1` VALUES LESS THAN (200),\n" +
+				" PARTITION `pMax` VALUES LESS THAN (MAXVALUE))"))
+		case model.StateNone.String():
+		default:
+			require.Failf(t, "unhandled schema state '%s'", schemaState)
+		}
+	}
+	postFn := func(tkO *testkit.TestKit, store kv.Storage) {
+		tkO.MustQuery(`select * from t where b = 5`).Sort().Check(testkit.Rows("5 5"))
+		tkO.MustQuery(`select * from t where b = "5"`).Sort().Check(testkit.Rows("5 5"))
+		tkO.MustExec(`admin check table t`)
+		tkO.MustQuery(`select * from t`).Sort().Check(testkit.Rows("1 1", "10 10", "101 101", "102 102", "11 11", "12 12", "13 13", "14 14", "2 2", "5 5", "6 6", "7 7", "8 8", "9 9", "984 984", "985 985", "986 986", "987 987", "988 988", "989 989", "990 990", "991 991", "992 992", "993 993", "998 998", "999 999"))
+		// TODO: Verify that there are no KV entries for old partitions or old indexes!!!
+		gcWorker, err := gcworker.NewMockGCWorker(store)
+		require.NoError(t, err)
+		require.Nil(t, gcWorker.DeleteRanges(context.TODO(), math.MaxInt64))
+		delRange := tkO.MustQuery(`select * from mysql.gc_delete_range_done`).Rows()
+		s := ""
+		for _, row := range delRange {
+			if s != "" {
+				s += "\n"
+			}
+			for i, col := range row {
+				if i != 0 {
+					s += " "
+				}
+				s += col.(string)
+			}
+		}
+		logutil.BgLogger().Info("gc_delete_range_done", zap.String("rows", s))
+		tkO.MustQuery(`select * from mysql.gc_delete_range`).Check(testkit.Rows())
+		ctx := tkO.Session()
+		is := domain.GetDomain(ctx).InfoSchema()
+		tbl, err := is.TableByName(context.Background(), pmodel.NewCIStr("test"), pmodel.NewCIStr("t"))
+		require.NoError(t, err)
+		tableID = tbl.Meta().ID
+		// Save this for the fix of https://github.com/pingcap/tidb/issues/56822
+		//GlobalLoop:
+		//	for _, globIdx := range originalGlobalIndexIDs {
+		//		for _, idx := range tbl.Meta().Indices {
+		//			if idx.ID == globIdx {
+		//				continue GlobalLoop
+		//			}
+		//		}
+		//		// Global index removed
+		//		require.False(t, HaveEntriesForTableIndex(t, tkO, tableID, globIdx), "Global index id %d for table id %d has still entries!", globIdx, tableID)
+		//	}
+	LocalLoop:
+		for _, locIdx := range originalIndexIDs {
+			for _, idx := range tbl.Meta().Indices {
+				if idx.ID == locIdx {
+					continue LocalLoop
+				}
+			}
+			// local index removed
+			for _, part := range tbl.Meta().Partition.Definitions {
+				require.False(t, HaveEntriesForTableIndex(t, tkO, part.ID, locIdx), "Local index id %d for partition id %d has still entries!", locIdx, tableID)
+			}
+		}
+		// TODO: Fix cleanup issues, most likely it needs one more SchemaState in onReorganizePartition
```

Comment:
Also disabled due to https://github.com/pingcap/tidb/issues/56822#issuecomment-2436343806.